File: unikqprocess.cpp
#include "unikqprocess.h"

UnikQProcess::UnikQProcess(QObject *parent) : QProcess(parent)
{
    connect(this, SIGNAL(readyReadStandardOutput()), this, SLOT(logOutProcess()));
    connect(this, SIGNAL(readyReadStandardError()), this, SLOT(logOutProcessErr()));
}

void UnikQProcess::run(const QByteArray cmd)
{
    run(cmd, false);
}

void UnikQProcess::run(const QByteArray cmd, bool detached)
{
    if(!detached){
         start(cmd);
    }else{
        startDetached(cmd);
    }
}

void UnikQProcess::logOutProcess()
{
    //QString data;
    //data.append(this->readAllStandardOutput());
    //qDebug()<<"------------>"<<data;
    setLogData(this->readAllStandardOutput());
}
void UnikQProcess::logOutProcessErr()
{
    setLogData(this->readAllStandardError());
}

File: UnikQProcess.py
from PySide2.QtCore import Property, QCoreApplication, QObject, Signal, Slot
import subprocess

class UnikQProcess(QObject):
    logDataChanged = Signal()
    def __init__(self, parent=None):
        super().__init__(parent)
        self._logData = ''


    @Property(str)
    def logData(self):
            return self._logData
    @Signal
    def logDataChanged(self):
        print('Signal lodData!')

    @Slot(result=str)
    def getLogData(self):
        return str(self._logData)
    @Slot(str)
    def setLogData(self, ld):
        self._logData=str(ld)
        #print('--->:::-->'+str(logData))
        self.logDataChanged.emit()

    @Slot(str)
    def run(self, cmd):
        listaCmd=cmd.split(sep=' ')
        command = subprocess.run(listaCmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding='UTF-8')
        out=str(command.stdout)
        #print(out)
        self.setLogData(out)

    logData = Property(str, getLogData, setLogData, notify=logDataChanged)

